d2lib/d2.go

```diff
@@ -68,14 +68,27 @@ func compile(ctx context.Context, g *d2graph.Graph, opts *CompileOptions) (*d2ta
 			return nil, err
 		}
 
-		constantNears := d2near.WithoutConstantNears(ctx, g)
+		constantNears, constantNearGraphs := d2near.WithoutConstantNears(ctx, g)
```

Comment:
do you need two variables? wouldn't constantNears just be the roots of the constantNearGraphs?

Reply:
fixed, and regerated some testcases

Reply:
@alixander i found that the sequence of keys of golang map is uncertain, which may need an extrat sorting, do you think it's better than two variables?

Reply:
can `constantNearGraphs` be an array then?

Reply:
yep, quite strightforward..., just need to use `tempGraph.Root.ChildrenArray[0]` to retrieve the near object. fixed